pkg/provider/aws/execution_unit.go

```diff
@@ -49,6 +49,12 @@ func (a *AWS) GenerateExecUnitResources(unit *core.ExecutionUnit, result *core.C
 	switch execUnitCfg.Type {
 	case Lambda:
 		lambdaFunction := resources.NewLambdaFunction(unit, a.Config.AppName, role, image)
+		if resources.VpcExists(dag) {
```

Comment:
Would this have a race condition with whether the VPC was created first or not? Or is there something that always sets the VPC up first before iterating through the resources.

Reply:
yeah this would if it required to proxy to an eks exec unit, but not for persist resources. I think we may need to determine if we are creating a vpc first thing

Reply:
That's fine then. Can you add a comment explaining why this is safe?

Reply:
i added the method to just check if it needs to be created and do that first instead just because its clearer and it may end up catching some other cases we dont think about